test/22_os-config.ts

```diff
@@ -0,0 +1,24 @@
+import { expect } from 'chai';
+
+import { supertest } from './test-lib/supertest';
+import { LOGS_HOST } from '../src/lib/config';
+
+describe('OS configuration endpoints', () => {
+	describe('/os/v1/config', () => {
+		it('should return a valid JSON response', async () => {
+			const { body } = await supertest().get('/os/v1/config').expect(200);
+
+			expect(body)
+				.to.have.property('services')
+				.that.has.all.keys('openvpn', 'ssh');
+			expect(body).to.have.property('schema_version').that.equals('1.0.0');
+			expect(body)
+				.to.have.property('config')
+				.that.deep.equals({
+					overrides: {
+						logsEndpoint: LOGS_HOST ? `https://${LOGS_HOST}` : undefined,
```

Comment:
Does this test pass if LOGS_HOST isn't defined?
I would expect it to error w/ something like `expected { logEndpoint: undefined } but got {}`.
A way to make this test both cases could be to have two separate `describe`s and changing the LOGS_HOST value in a `before` in each of them.
See: https://github.com/balena-io/balena-api/blob/7e6d06fe8e75d7aee5a0f0079e0f21794acc7e14/test/01_user.ts#L265-L267

Reply:
This test does indeed pass even with `LOGS_HOST` undefined, with overrides being equal to `{}`. 

Do you feel strongly about adding test mock env var handling to the open-balena-api codebase? It doesn't feel entirely comfortable to me as it's a branch in the production code written specifically for tests. If you do feel strongly, I'm fine with it @thgreasi :)

Reply:
@thgreasi friendly reminder :) if you don't feel strongly about this, I'm going to go ahead and resolve both comments in 24h and merge